### PR TITLE
modal unit tests

### PR DESCRIPTION
**Pull request by vxsx**
- this PR adds some tests for cms.modal
- moves modal declaration out of document ready
- fixes tests for cms.base (apparently reassigning window.parent is not really possible in real environments)
- modifies karma config to work correctly with saucelabs on a limited amount of browsers
- adds other static files to what karma serves

so far it brings us to a 37% coverage for modal
screen shot 2015-12-09 at 10 40 30

locally i've tested with
screen shot 2015-12-09 at 14 54 00

some tests for modal public methods are still missing, but they will come

### CODE REVIEW

File: cms/static/cms/js/modules/cms.base.js [hunk 1]
```diff
@@ -101,7 +101,8 @@ var CMS = {
         reloadBrowser: function (url, timeout, ajax) {
             var that = this;
             // is there a parent window?
-            var parent = (window.parent) ? window.parent : window;
+            var win = this._getWindow();
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.base.js]: isn't this a bit overkill?

File: cms/tests/frontend/karma.conf.js [hunk 2]
```diff
@@ -130,15 +123,8 @@ module.exports = function (config) {
...
-            // travis
-            if (useSauceLabs()) {
-                return 3;
-            } else {
-                return Infinity;
-            }
-        })(),
-        browserNoActivityTimeout: 50 * 1000,
+        concurrency: Infinity,
+        browserNoActivityTimeout: 2 * 60 * 1000,
```
Comment (FinalAngel) [inline cms/tests/frontend/karma.conf.js]: could we document why it's like this?

File: cms/tests/frontend/unit/cms.modal.test.js [hunk 3]
```diff
@@ -3,17 +3,343 @@
...
+        expect(CMS.Modal.prototype.minimize).toEqual(jasmine.any(Function));
+        expect(CMS.Modal.prototype.maximize).toEqual(jasmine.any(Function));
+    });
+
+    describe('instance', function () {
+        it('has ui', function (done) {
+            $(function () {
+                var modal = new CMS.Modal();
+                expect(modal.ui).toEqual(jasmine.any(Object));
+                expect(Object.keys(modal.ui)).toContain('modal');
```
Comment (FinalAngel) [inline cms/tests/frontend/unit/cms.modal.test.js]: might be easier to do a loop here?

File: cms/tests/frontend/unit/cms.base.test.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-/* globals jQuery, Class, $, document, window, localStorage, pending */
```
Comment (FinalAngel) [inline cms/tests/frontend/unit/cms.base.test.js]: aren't these all unrelated changes to the modal?

Comment (vxsx) [inline cms/tests/frontend/unit/cms.base.test.js]: yup

File: cms/tests/frontend/unit/cms.modal.test.js (hunk 3 above)
Comment (vxsx) [inline cms/tests/frontend/unit/cms.modal.test.js]: can you elaborate? 
object.keys order is not enforced

File: cms/static/cms/js/modules/cms.base.js (hunk 1 above)
Comment (vxsx) [inline cms/static/cms/js/modules/cms.base.js]: nope

File: cms/tests/frontend/karma.conf.js (hunk 2 above)
Comment (vxsx) [inline cms/tests/frontend/karma.conf.js]: sure

File: cms/tests/frontend/unit/cms.modal.test.js (hunk 3 above)
Comment (FinalAngel) [inline cms/tests/frontend/unit/cms.modal.test.js]: nvm

File: cms/tests/frontend/unit/cms.modal.test.js [hunk 5]
```diff
@@ -3,17 +3,447 @@
...
+            expect(css).toEqual(jasmine.any(Object));
+            expect(Object.keys(css)).toContain('margin-left');
+            expect(Object.keys(css)).toContain('margin-top');
+            expect(Object.keys(css)).toContain('top');
+            expect(Object.keys(css)).toContain('left');
+
+            modal.minimize(); // restore
+        });
+
+        it('doesnt minimize maximized modal', function () {
```
Comment (FinalAngel) [inline cms/tests/frontend/unit/cms.modal.test.js]: how do we handle language here? "doesn't"?

**Comment by FinalAngel** [comment]
sweet work

File: cms/tests/frontend/unit/cms.modal.test.js (hunk 5 above)
Comment (vxsx) [inline cms/tests/frontend/unit/cms.modal.test.js]: should be `does not` :D

**Comment by FinalAngel** [comment]
LGTM
